### ISSUE TITLE
[SR-13882] oldValue in didSet is reported as nil by debugger

### ISSUE THREAD

**Issue by idrougge** [swiftlang/llvm-project#4326]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-13882      |
   |Radar             | rdar://problem/71677710         |
   |Original Reporter | @idrougge      |
   |Type              | Bug    |
   |Status            | Resolved        |
   |Resolution        | Done    |
        
<details>
  <summary>Environment</summary>

Xcode 12, Xcode 12.1

</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | LLDB for Swift    |
   |Labels            | Bug        |
   |Assignee          | @adrian-prantl      |
   |Priority          | Medium      |

   

   md5: 665d70b7dd908130d8aa01a7703f067f

  </details>

**Issue Description:**

\`oldValue\` in a \`didSet\` property observer is represented as \`nil\` when setting a breakpoint inside the \`didSet\` observer, when printing the local variables using \`(lldb) frame variable\` or using Xcode's "Variables" debugger view. \`po oldValue\` will however print out a non-nil value.

This behaviour only seems to occur when the property observed is a struct (not a class) containing an optional value of a certain type, such as \`Date\`.

A code example follows:

```` java
```
import Foundation

struct Slot: Equatable {
    let time: Date?
}

class Handler {
    var slot: Slot? {
        didSet {
            guard
                slot != oldValue // breakpoint here
            else {
                return
            }
        }
    }
    
    init() {
        self.slot = .init(time: Date()) // slot is set already in init, so oldValue cannot be nil when breakpoint is triggered
    }
}

let handler = Handler()
handler.slot = .init(time: .distantFuture)
handler.slot = .init(time: .distantPast)
```
````

**Comment by typesanitizer** [comment]
@swift-ci create

**Comment by augusto2112** [comment]
@adrian-prantl this bug occurs on lldb shipped with macOS, but is already fixed on the main branch. Can we close it?

**Comment by adrian-prantl** [comment]
According Augusto's comment this is already fixed!
